api/server_test.go

```diff
@@ -101,24 +109,119 @@ func TestRatsdChares_missing_nonce(t *testing.T) {
 	assert.Equal(t, expectedBody, &body)
 }
 
-func TestRatsdChares_valid_request(t *testing.T) {
+func TestRatsdChares_valid_request_no_available_attester(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+
 	var params RatsdCharesParams
 
 	param := fmt.Sprintf(`application/eat+jwt; eat_profile=%q`, TagGithubCom2024Veraisonratsd)
 	params.Accept = &param
 	logger := log.Named("test")
-	s := &Server{logger: logger}
+
+	pluginList := []string{}
+	dm := mock_deps.NewMockIManager(ctrl)
+	dm.EXPECT().GetPluginList().Return(pluginList)
+
+	s := NewServer(logger, dm)
 	w := httptest.NewRecorder()
-	rb := strings.NewReader("{\"nonce\": \"MIDBNH28iioisjPy\"}")
+	rs := fmt.Sprintf("{\"nonce\": \"%s\"}", validNonce)
+	rb := strings.NewReader(rs)
 	r, _ := http.NewRequest(http.MethodPost, "/ratsd/chares", rb)
 	r.Header.Add("Content-Type", ApplicationvndVeraisonCharesJson)
 	s.RatsdChares(w, r, params)
 
-	expectedCode := http.StatusOK
-	expectedType := param
-	expectedBody := "hello from ratsd!"
+	expectedCode := http.StatusInternalServerError
+	expectedType := problems.ProblemMediaType
+	expectedDetail := "no sub-attester available"
+	expectedBody := problems.NewDetailedProblem(http.StatusInternalServerError, expectedDetail)
+
+	var body problems.DefaultProblem
+	_ = json.Unmarshal(w.Body.Bytes(), &body)
 
 	assert.Equal(t, expectedCode, w.Code)
 	assert.Equal(t, expectedType, w.Result().Header.Get("Content-Type"))
-	assert.Equal(t, expectedBody, w.Body.String())
+	assert.Equal(t, expectedBody, &body)
+}
+
+func TestRatsdChares_valid_request(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+
+	var params RatsdCharesParams
+
+	param := fmt.Sprintf(`application/eat+jwt; eat_profile=%q`, TagGithubCom2024Veraisonratsd)
+	params.Accept = &param
+	logger := log.Named("test")
+
+	pluginList := []string{"mock-tsm"}
+	dm := mock_deps.NewMockIManager(ctrl)
+	dm.EXPECT().GetPluginList().Return(pluginList).AnyTimes()
+	dm.EXPECT().LookupByName("mock-tsm").Return(mocktsm.GetPlugin(), nil).AnyTimes()
+
+	s := NewServer(logger, dm)
+	realNonce, _ := base64.RawURLEncoding.DecodeString(validNonce)
+
+	tests := []struct {
+		name, query string
+		privlevel   int
+	}{
+		{
+			"no params",
+			fmt.Sprintf("{\"nonce\": \"%s\"}", validNonce),
```

Comment:
You can avoid escaping by using "backtick-strings”:

```go

	testMetaValid   = []byte(`{
		"signer": {
			"name": "ACME Ltd signing key",
			"uri": "https://acme.example"
		},
		"validity": {
			"not-before": "2021-12-31T00:00:00Z",
			"not-after": "2025-12-31T00:00:00Z"
		}
	}`)


```

Reply:
resolved.